File: container_rental/container_rental/report/overdue_containers_report/overdue_containers_report.py
# S11 export twin — shares the exact query of the overdue-containers page.
from frappe import _

from container_rental.api import get_overdue_data


def execute(filters=None):
	return get_columns(), get_overdue_data(filters or {})


def get_columns():
	return [
		{"fieldname": "container", "label": _("رقم الحاوية"), "fieldtype": "Link", "options": "Container", "width": 110},
		{"fieldname": "client_name", "label": _("العميل"), "fieldtype": "Data", "width": 150},
		{"fieldname": "mobile_no", "label": _("رقم الجوال"), "fieldtype": "Data", "width": 110},
		{"fieldname": "container_size", "label": _("الحجم"), "fieldtype": "Data", "width": 90},
		{"fieldname": "classification", "label": _("التصنيف"), "fieldtype": "Data", "width": 100},
		{"fieldname": "branch", "label": _("الفرع"), "fieldtype": "Data", "width": 110},
		{"fieldname": "address", "label": _("العنوان / الموقع"), "fieldtype": "Data", "width": 180},
		{"fieldname": "driver_name", "label": _("السائق"), "fieldtype": "Data", "width": 130},
		{"fieldname": "delivered_on", "label": _("تاريخ التوصيل"), "fieldtype": "Datetime", "width": 140},
		{"fieldname": "due_on", "label": _("تاريخ الاستحقاق"), "fieldtype": "Datetime", "width": 140},
		{"fieldname": "overdue_text", "label": _("مدة التأخير"), "fieldtype": "Data", "width": 140},
		{"fieldname": "severity_label", "label": _("درجة الخطورة"), "fieldtype": "Data", "width": 110},
		{"fieldname": "last_whatsapp_message", "label": _("آخر رسالة واتساب"), "fieldtype": "Data", "width": 130},
		{"fieldname": "last_whatsapp_on", "label": _("تاريخ آخر رسالة"), "fieldtype": "Datetime", "width": 140},
	]

File: container_rental/container_rental/report/overdue_containers_report/overdue_containers_report.js
frappe.query_reports["Overdue Containers Report"] = {
	filters: [
		{
			fieldname: "classification",
			label: __("التصنيف"),
			fieldtype: "Link",
			options: "Container Classification",
		},
		{
			fieldname: "container_size",
			label: __("حجم الحاوية"),
			fieldtype: "Link",
			options: "Container Size",
		},
		{ fieldname: "branch", label: __("الفرع"), fieldtype: "Link", options: "Rental Branch" },
		{
			fieldname: "driver",
			label: __("السائق"),
			fieldtype: "Link",
			options: "Employee",
			get_query: () => ({ filters: { designation: "سائق" } }),
		},
		{
			fieldname: "delay_range",
			label: __("نطاق مدة التأخير"),
			fieldtype: "Select",
			options: ["", "0-2", "3-7", "7+"],
		},
	],

	formatter(value, row, column, data, default_formatter) {
		let html = default_formatter(value, row, column, data);
		if (data && data.severity) {
			const colors = { yellow: "#b58a00", orange: "#d35400", red: "#c0392b" };
			if (column.fieldname === "overdue_text" || column.fieldname === "severity_label") {
				html = `<span style="color:${colors[data.severity]};font-weight:bold">${value || ""}</span>`;
			}
		}
		return html;
	},
};
